rustls/src/quic.rs

```diff
@@ -130,27 +222,25 @@ pub struct Keys {
 
 impl Keys {
     /// Construct keys for use with initial packets
-    pub fn initial(
-        initial_salt: &hkdf::Salt,
-        client_dst_connection_id: &[u8],
-        is_client: bool,
-    ) -> Self {
+    pub fn initial(initial_salt: &[u8], client_dst_connection_id: &[u8], is_client: bool) -> Self {
```

Comment:
Since we've already added a `Version` enum, should we bake in the initial salt as well based on it?

Reply:
The initial salt really requires more granularity than the versions, right?

Reply:
I think if there's a way to eliminate all (any) potential for error in the choosing of the initial salt, we should do so now.

IIUC, the spec says the salt will be 0x38762cf7f55934b34d179ae6a4c80cadccbb7f0a so I'm not sure why we're asking the caller to pass it in, instead of just using that value as a constant.

Reply:
The initial salt cannot be updated without a version change. There were multiple salts pre-1.0, but I don't think we need to support more than the latest draft salt, since it hadn't changed for some time at that point, and at least quinn isn't going to make any effort to support very old draft versions.

Reply:
So do we decide to only support draft 29-32 and RFC 9001?

* Draft [23, 28]: 0xc3eef712c72ebb5a11a7d2432bb46365bef9f502
* Drafts [29, 32]: 0xafbfec289993d24c9e9786f19c6111e04390a899
* Drafts 33, 34, RFC 9001: 0x38762cf7f55934b34d179ae6a4c80cadccbb7f0a

Reply:
> So do we decide to only support draft 29-32 and RFC 9001?

Why even bother continuing to supporting older drafts. I suggest we just support the final version.

Reply:
I don't have a strong opinion either way, but if we drop draft version salt support we should also strip out the draft TLS extension number support. Maybe keep the version enum/interface around, since there's talk of an early, trivial QUIC 2 for anti-ossification purposes

Reply:
As I understand from the QUIC mailing list, draft 29 will be around for a bit at least in browsers.

Reply:
I implemented this, keeping `V1Draft` and documenting it explicitly as supporting drafts 29 through 32.